### ISSUE TITLE
[Bug]: ObjectExpected, Wrap finalizer for PromiseRaw failed

### ISSUE THREAD

**Issue by emielwit** [web-infra-dev/rsbuild#4056]
### Version

```sh
System:
  OS: Linux 5.15 Debian GNU/Linux 11 (bullseye) 11 (bullseye)
  CPU: (8) x64 Intel(R) Xeon(R) Platinum 8171M CPU @ 2.60GHz
  Memory: 790.60 MB / 31.34 GB
  Container: Yes
  Shell: 5.1.4 - /bin/bash
npmPackages:
  @rsbuild/core: 1.1.5 => 1.1.5 
  @rsbuild/plugin-node-polyfill: 1.2.0 => 1.2.0 
  @rsbuild/plugin-react: 1.0.6 => 1.0.6
```

### Details

After updating from 1.0.19 to 1.1.5 (also 1.1.2) we got the following error during build:
![Image](https://github.com/user-attachments/assets/7014bf05-680f-4806-a701-124794d4db7c)

### Reproduce link

https://codesandbox.io/p/github/rspack-contrib/rsbuild-codesandbox-example/csb-9ztqcd/draft/solitary-currying

### Reproduce Steps

1. Fork codesandbox
2. run bun install
3. run bun run build
4. error should be raised

It looks like if I only use bun and there is no nodejs it fails.

**Comment by chenjiahan** [comment]
Maybe a bug of bun: https://github.com/oven-sh/bun/issues/15383

**Comment by chenjiahan** [comment]
Considering this is probably a Bun issue, I will close the current issue.

If you find this issue is related to Rspack/Rsbuild, feel free to reopen it.

**Comment by emielwit** [comment]
Resolved in Bun 1.1.39
